library/core/src/main/java/com/google/android/exoplayer2/extractor/flv/FlvExtractor.java

```diff
@@ -110,6 +110,7 @@ public boolean sniff(ExtractorInput input) throws IOException, InterruptedExcept
     input.peekFully(scratch.data, 0, 4);
     scratch.setPosition(0);
     int dataOffset = scratch.readInt();
+    if (dataOffset > 9) dataOffset = 9;
```

Comment:
We allow streams from any method, which would commonly include android, iphone, obs, xsplit, tricaster, vidiu, etc. We've had a commit like this in place for a long time now so I'll see if I have any previous records. @rkrishnan2012 may be able to provide more context as he is the one that worked on the fix originally. I'm not sure if these videos would end up playing or not. I do know #2688 was the more prevalent OOM cause.